lib/src/blocs/authentication_blocs/authentication_blocs.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '/src/blocs/authentication_blocs/authentication_event.dart';
+import '/src/repositories/user_repository.dart';
+
+import 'authentication_state.dart';
+
+class AuthenticationBloc
+    extends Bloc<AuthenticationEvent, AuthenticationState> {
+  final UserRepository _userRepository;
+
+  AuthenticationBloc({required UserRepository userRepository})
+      : _userRepository = userRepository,
+        super(AuthenticationInitial());
+
+  @override
+  Stream<AuthenticationState> mapEventToState(
+      AuthenticationEvent event) async* {
+    if (event is AuthenticationStarted) {
+      yield* _mapAuthenticationStartedToState();
+    } else if (event is AuthenticationLoggedIn) {
+      yield* _mapAuthenticationLoggedInToState();
+    } else if (event is AuthenticationLoggedOut) {
+      yield* _mapAuthenticationLoggedOutInToState();
+    }
+  }
```

Comment:
=> switch case

Reply:
Right now i designed it for next and currently it is not used in this project